src/macsniff.cpp

```diff
@@ -0,0 +1,145 @@
+
+// Basic Config
+#include "main.h"
+#include "globals.h"
+
+#ifdef BLECOUNTER
+#include <BLEDevice.h>
+#include <BLEUtils.h>
+#include <BLEScan.h>
+#include <BLEAdvertisedDevice.h>
+#endif
+
+#ifdef VENDORFILTER
+  #include <array>
+  #include <algorithm>
+  #include "vendor_array.h"
+#endif
+
+// Local logging tag
+static const char *TAG = "macsniff";
+
+static wifi_country_t wifi_country = {.cc=WIFI_MY_COUNTRY, .schan=WIFI_CHANNEL_MIN, .nchan=WIFI_CHANNEL_MAX, .policy=WIFI_COUNTRY_POLICY_MANUAL};
+
+uint16_t currentScanDevice = 0;
+
+bool mac_add(uint8_t *paddr, int8_t rssi, bool sniff_type) {
+
+    char counter [11]; // uint32_t -> 4 byte -> 10 decimals + '0' terminator -> 11 chars
+    char macbuf [21]; // uint64_t -> 8 byte -> 20 decimals + '0' terminator -> 21 chars
+    char typebuff[8] ;
+    bool added = false;
```

Comment:
do we need this? bool state of insertion is given by insert statement (see below).

Reply:
no it's not needed, sure

Reply:
i was wrong. It's needed, otherwise the log messages are not appropriate. changed in v1.2.88.